enterprise/models.py

```diff
@@ -1015,14 +1030,30 @@ def __repr__(self):
         """
         return self.__str__()
 
+    @property
+    def identity_provider(self):
+        """
+        Associated identity provider instance.
+        """
+        identity_provider = utils.get_identity_provider(self.provider_id)
+        return identity_provider
+
     @property
     def provider_name(self):
         """
         Readable name for the identity provider.
         """
-        identity_provider = utils.get_identity_provider(self.provider_id)
+        identity_provider = self.identity_provider
         return identity_provider and identity_provider.name
 
+    @property
+    def sync_learner_profile_data(self):
+        """
+        Readable name for the identity provider.
+        """
+        identity_provider = self.identity_provider
+        return identity_provider is not None and identity_provider.sync_learner_profile_data
```

Comment:
Q: Is there a test that tests the 2nd half of the `and` clause?

Reply:
Nope, do we need one?